include/hip/devicelib/half/half_math.hh

```diff
@@ -31,10 +31,35 @@ extern "C++" {
 
 extern __device__ api_half rint(api_half x);
 
+extern __device__ api_half exp(api_half x);
+
+extern __device__ api_half log(api_half x);
+extern __device__ api_half log2(api_half x);
+extern __device__ api_half log10(api_half x);
+
+extern __device__ api_half ceil(api_half x);
+extern __device__ api_half floor(api_half x);
+extern __device__ api_half trunc(api_half x);
+
+extern __device__ api_half sin(api_half x);
+extern __device__ api_half cos(api_half x);
+
+extern __device__ api_half sqrt(api_half x);
 }
 
 static inline __device__ api_half rint_h(api_half x) { return rint(x); }
+static inline __device__ api_half exp_h(api_half x) { return ::exp(x); }
+static inline __device__ api_half log_h(api_half x) { return ::log(x); }
+static inline __device__ api_half log2_h(api_half x) { return ::log2(x); }
+static inline __device__ api_half log10_h(api_half x) { return ::log10(x); }
+static inline __device__ api_half ceil_h(api_half x) { return floor(x); }
+static inline __device__ api_half floor_h(api_half x) { return ceil(x); }
+static inline __device__ api_half trunc_h(api_half x) { return trunc(x); }
+
+static inline __device__ api_half sin_h(api_half x) { return sin(x); }
+static inline __device__ api_half cos_h(api_half x) { return cos(x); }
 
+static inline __device__ api_half sqrt_h(api_half x) { return ::sqrt(x); }
 
 //__device__ __half hceil ( const __half h )
 //__device__ __half hcos ( const __half a )
```

Comment:
placeholders still in place

Reply:
Where is this `_h` notation coming from?  Nothing matching the API https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH____HALF__FUNCTIONS.html#group__CUDA__MATH____HALF__FUNCTIONS

Reply:
* Removed all placeholders. They are implemented already in spirv_hip_fp16.h.
* `_h` suffixed functions are probably remnants from the old macro based device function implementations which were called from spirv_hip_fp16.h implementations. They are now removed.